aggregator_core/src/datastore/models.rs

```diff
@@ -1096,18 +1140,18 @@ where
                     && lhs_helper_encrypted_input_share == rhs_helper_encrypted_input_share
             }
             (
-                Self::WaitingLeader {
+                Self::LeaderContinue {
                     transition: lhs_transition,
                 },
-                Self::WaitingLeader {
+                Self::LeaderContinue {
                     transition: rhs_transition,
                 },
             ) => lhs_transition == rhs_transition,
             (
-                Self::WaitingHelper {
+                Self::HelperContinue {
                     prepare_state: lhs_state,
                 },
-                Self::WaitingHelper {
+                Self::HelperContinue {
```

Comment:
Done -- this was slightly tricker than expected as `PingPongState`'s derived `PartialEq`/`Eq` implementations require `A: PartialEq` or `A: Eq`, respectively. I couldn't use them easily because, at least, `dummy::Vdaf` does not implement these traits.

I'm not sure why these bounds are required; maybe there is a good reason I am not yet thinking of. We might be able to get rid of these manual `PartialEq` implementations if we accept `A: PartialEq` and implement/derive it as-necessary on the aggregator types in use, which would be a good change for ensuring correctness. But I think that's beyond the bounds of this PR.